docs:  Add additional information and tips to Backup & Restore Usage Documentation Page & Shopping List

ChipSkylark37 opened the pull request:
## What type of PR is this?

- documentation

<!--
  Delete any of the following that do not apply:
 -->




## What this PR does / why we need it:

- Adding notes regarding the Settings > Backup page in addition to the URL
- Listed out some additional actions that can be performed in this page. Added a tip for manual review / adjustment of the database.json file in the event that a restore fails
- Changed some verbiage (removing pluralization on some items)

<!--
  What goal is this change working towards?
  Provide a bullet pointed summary of how each file was changed.
  Briefly explain any decisions you made with respect to the changes.
  Include anything here that you didn't include in *Release Notes*
  above, such as changes to CI or changes to internal methods.

  If there is a UI component to the change, please include before/after images.
-->

## Which issue(s) this PR fixes:

None - I am looking through some of the documentation and looking to offer some assistance on pieces that could use clarity or an update. 

<!--
If this PR fixes one of more issues, list them here.
One per line, like so:
Fixes #123
Fixes #39
-->

## Special notes for your reviewer:

This is my first PR so I wanted to make some small changes to the documentation and see what the process looks like.

<!--
   Is there any particular feedback you would / wouldn't like?
   Which parts of the code should reviewers focus on?
-->

ChipSkylark37 committed:
Update backups-and-restoring.md

- Adding notes regarding the Settings > Backup page in addition to the URL
- Listed out some additional actions that can be performed in this page.
Added a tip for manual review / adjustment of the database.json file in the event that a restore fails
- Changed some verbiage (removing pluralization on some items)

boc-the-git commented:
Thanks for taking the time to raise a PR @ChipSkylark37!

Whilst I haven't gone through the actual doco change in detail, there's 2 key things to address with the PR itself.
1. You'll see that the CI checks are failing, that'll be because `DOCS` is not an allowable prefix for the PR title - `docs` is what you'd be after.
2. The rest of the title itself could be more meaningful. Something generic like `Update backups-and-restoring.md` isn't overly useful to someone reading a changelog whereas (for example) something like `Added additional information to the backup/restore documentation` is a bit more meaningful.

ChipSkylark37 committed:
Merge branch 'mealie-recipes:mealie-next' into patch-1

ChipSkylark37 committed:
Add additional information and tips to Backup & Restore Usage

- Adding notes regarding how to access via the Settings > Backup page in addition to the URL slug.
- Location of 'backup button' changed from bottom left to top left
 - Listed out some additional actions that can be performed in this page. 
- Added a tip for manual review / adjustment of the database.json file in the event that a restore fails
- Changed some verbiage (removing pluralization on some items)

ChipSkylark37 commented:
I believe I have corrected the errors and performed a new commit after syncing the latest nightly.

ChipSkylark37 committed:
Update Shopping Lists to include additional actions able to be performed, tips, and some examples.

- Added location of Shopping Lists from Sidebar.
- Added some common things that can be done (See added items, see linked recipes, checking off an item, sorting items, creating / sorting labels.
- Added tip for unchecking a checked of item
- Added tip for using 'Food' / 'Smart Handling' to combine ingredients
- Added tip for using Labels to sort (however you may want)

ChipSkylark37 commented:
Additionally I made some changes to the Shopping List Section.

Would it be recommended to start a new commit for each one of these? I'm still learning the system a bit so I could use some advice on that.

ChipSkylark37 committed:
Merge branch 'mealie-next' into patch-1

ChipSkylark37 commented:
Hey all - I was just recently moving my instance from SQL > Postgres and was going to update the documentation a bit more. 

I synced the latest mealie:nightly to my own fork, which overwrote the documentation changes in my local fork.

If this PR gets merged, I can then resync and do some further updates.

Is there anything I need to do to help get these changes merged?

boc-the-git commented:
> Would it be recommended to start a new commit for each one of these?

Depends if you mean commit or pull request.
Generally, yes it's best to have different commits for unrelated changes - though in the case of docs that isn't critically important.

As for a pull request, the general rule is that the smaller the better, as it allows for it to be reviewed more easily. Again for docs though, it's a grey area.

boc-the-git commented:
> I synced the latest mealie:nightly to my own fork, which overwrote the documentation changes in my local fork.

They'd still exist on the branch they belong to - you can see them on here: https://github.com/ChipSkylark37/mealie/tree/patch-1

boc-the-git commented on docs/docs/documentation/getting-started/features.md:
```diff
@@ -84,7 +84,27 @@ The meal planner has the concept of plan rules. These offer a flexible way to us
+- See linked recipes with ingredients
+    - Toggling via the 'Pot' icon will show you the linked recipe, allowing you to click to access it. 
+- Check off an item
+- Add / Change / Remove / Sort Items via the '3x3' grid icon
```
I'm not certain what is meant by this. Perhaps you're referring to the button that is 3x4?

ChipSkylark37 replied:
Yes, I didn't realize it was 3x4 since most of the grids that I come across are 3x3. Is there a formal name for that button that I could use instead?

ChipSkylark37 replied:
I changed to 3x4 and also included an image.

boc-the-git commented on docs/docs/documentation/getting-started/features.md:
```diff
@@ -84,7 +84,27 @@ The meal planner has the concept of plan rules. These offer a flexible way to us
+Managing shopping lists can be done from the Sidebar > Shopping Lists. 
+
+Here you will be able to:
+- See list of already added items to the Shopping List
```
```suggestion
- See items already on the Shopping List
```

boc-the-git commented on docs/docs/documentation/getting-started/features.md:
```diff
@@ -84,7 +84,27 @@ The meal planner has the concept of plan rules. These offer a flexible way to us
+    - 'No Label' will always be on the top, others can be Reordered via the 'Reorder Labels' button
 
+!!! tip
+    If you accidently checked off an item, you can uncheck it by expanding 'items checked' and unchecking it. This will automatically add it back to the Shopping List.
```
```suggestion
    If you accidentally checked off an item, you can uncheck it by expanding 'items checked' and unchecking it. This will add it back to the Shopping List.
```

It isn't really automatic, as you're taking manual action to do it.

ChipSkylark37 replied:
True. My thought was that it doesn't stay in the 'checked off' area and instead moves back to the top lists. I'll adjust the language.

boc-the-git commented on docs/docs/documentation/getting-started/features.md:
```diff
@@ -84,7 +84,27 @@ The meal planner has the concept of plan rules. These offer a flexible way to us
+    If you accidently checked off an item, you can uncheck it by expanding 'items checked' and unchecking it. This will automatically add it back to the Shopping List.
+    
+!!! tip
+    Along with Smart Handling, you can toggle 'Food' on individual items so that if you add multiple of the same food / ingredient, Mealie will automatically combine them together. Do this by editing an item in the Shopping List and clicking the 'apple' icon. If you added "1 | cup | cheese" and "2 | cup | cheese" this would be combined to show "3 cups of cheese" Please note this cannot be undone  ...
```
Without context, even I don't know what "Smart Handling" means here. Think that needs simplified for a new user.
I don't have exact wording, but maybe keeping this tip much simpler and linking people to the FAQ that guides them on how ingredients get combined.

ChipSkylark37 replied:
Updated to hopefully clarify and linking to FAQ section.

ChipSkylark37 committed:
Simplify Language for Items on Shopping List

I agree with the changes to make this read better.

Co-authored-by: boc-the-git <3479092+boc-the-git@users.noreply.github.com>

ChipSkylark37 committed:
Update Features for 'Unchecking' Shopping List Items

Accepted suggestion for how 'Unchecking' an item is handled.

Co-authored-by: boc-the-git <3479092+boc-the-git@users.noreply.github.com>

ChipSkylark37 committed:
Making Suggested Changes from PR

- Changed language as per suggestions: https://github.com/mealie-recipes/mealie/pull/4843
- Moved around some of the tips that were added
- Added a note on using a 'Want to Make' tag under Cookbooks as per a discord conversation with Michael (great tip!)

ChipSkylark37 reviewed (commented):
Reviewed and make changes per conversations.

ChipSkylark37 committed:
Merge branch 'mealie-next' into patch-1

ChipSkylark37 commented:
Forgive me as I don't want to spam, but I believe I've made all requested changes. 

Since I'm still new to this, I can't tell if there's any action needed on my part or if this is all the normal checks / waiting for someone else to review & approve.

Cheers!

Kuchenpirat commented:
If you have addressed the feedback above you should be all good to go.
We are a pretty small team with a lot of things going on outside of unpaid work for mealie, so things might take time.
As this is just a little docs pr where not much testing is needed, I expect it to be addressed sooner rather than later.

Kuchenpirat commented on docs/docs/documentation/getting-started/features.md:
```diff
@@ -84,7 +85,30 @@ The meal planner has the concept of plan rules. These offer a flexible way to us
+- See linked recipes with ingredients
+    - Toggling via the 'Pot' icon will show you the linked recipe, allowing you to click to access it. 
+- Check off an item
+- Add / Change / Remove / Sort Items via the grid icon (![image](https://github.com/user-attachments/assets/ad59a5ea-2ee5-4c6c-9002-9312f018290f))
```
We should not rely on github to host our images. 
You can use the `docs -> docs -> assets -> img` folder. 
The images also have a bit much margin towards the bottom making them stick out (see screeshot below)

![image](https://github.com/user-attachments/assets/47de1e08-9695-4e24-904e-fb0bd2679a12)

ChipSkylark37 replied:
As these appear to be SVG files, is there a way I can use that code to draw it vs uploading the image? Would that be preferred instead of a static image perhaps? 

For example:

![image](https://github.com/user-attachments/assets/80ecbabf-ffbd-4878-9217-422f97d351cf)

= 

![image](https://github.com/user-attachments/assets/6386180c-31ef-43c9-b65e-c5492f4cebdb)

ChipSkylark37 replied:
Removed images for now.

Kuchenpirat commented on docs/docs/documentation/getting-started/features.md:
```diff
@@ -84,7 +85,30 @@ The meal planner has the concept of plan rules. These offer a flexible way to us
+!!! tip
+    You can use Labels to categorize your ingredients. You may want to Label by Food Type (Frozen, Fresh, etc), by Store, Tool, Recipe, or more. Play around with this to see what works best for you.
+    
+!!! tip
+   You can toggle 'Food' on items so that if you add multiple of the same food / ingredient, Mealie will automatically combine them together. Do this by editing an item in the Shopping List and clicking the 'apple' icon (![image](https://github.com/user-attachments/assets/0925c88b-77aa-4446-8539-2ee436243e15)). If you then have recipes that contain "1 | cup | cheese" and "2 | cup | cheese" this w ...
```
This is not contained within the Admonitions box

ChipSkylark37 replied:
Sorry, whats the 'admonitions box' ?

Kuchenpirat replied:
It is the item you use to place your tips in (See docs [here ](https://squidfunk.github.io/mkdocs-material/reference/admonitions/))
Currently the text is not inside of the box so there is just an empty field that has text writen in it. 
I think you might have to fix the indentation. It differs from the other ones. 

Screenshot below. 

![image](https://github.com/user-attachments/assets/9989a628-ecef-4482-9929-b5b5b932eb2e)

ChipSkylark37 replied:
Oh gotcha, will do. Preview doesn't show me that for some reason... Perhaps I should be doing it in vscode instead so I can watch the changes live vs just Github Preview.

Kuchenpirat replied:
Ah ok, if you follow this guide you should be able to get a dev env started up pretty easily. 
then `task` will list all availlable commands and `task docs` will spin up a live version of the docs site locally.

ChipSkylark37 committed:
Removing Images / Updating Tip Indentation

- Removed Images until I can determine the best way to upload them (or use the SVG files)
- Fixed the 'four spaces' instead of using a Tab.

ChipSkylark37 reviewed (commented):
Reviewed suggested changes:
- (Hopefully) fixed the indentation. I used a 'Tab' instead of 'Four Spaces'
- Removed images / assets temporarily while I think about the best way to do this moving forward since I like the idea of images, but dont want to have to upload images each time.

Kuchenpirat committed:
Merge branch 'mealie-next' into patch-1

Kuchenpirat commented on docs/docs/documentation/getting-started/features.md:
```diff
@@ -62,6 +62,7 @@ Mealie also has the concept of cookbooks. These can be created inside of a group
 - Main Courses: This cookbooks has all the recipes that have the **Dinner** category
 - Pasta Sides: Recipes that have both the **Side** category and the **Pasta** tag
 - Dessert Breads: Recipes that have both the **Bread** category and the **Dessert** tag
+- Make Later: Recipes that have a **Want To Make** tag
```
I don't think that is a good example here, because you would want to remove that tag after you have made it.

ChipSkylark37 replied:
Maybe a note on there to remove it afterwards as this is actually one of my most used cookbooks currently as opposed to removing it? 

Other than a manual 'tag' search for that (which I would need to remove manually as well) - The cookbook is the fastest way to see that.

Kuchenpirat replied:
I use favorites or the meal planner for that, but I can see that use case.
Let's not add an additional note to that, just adds clutter

ChipSkylark37 committed:
Remove 'Make It Later' Tag

To avoid confusion / manual modifications, removing the note of "Make it Later'

ChipSkylark37 reviewed (commented):
removed 'Make Later'

Kuchenpirat reviewed (approved):
This is good to go in now🚀
Sorry it took me so long to get back to this.

Kuchenpirat committed:
Merge branch 'mealie-next' into patch-1

ChipSkylark37 commented:
Great thanks!
